htmresearch/algorithms/faulty_temporal_memory.py

```diff
@@ -70,206 +62,57 @@ def killCells(self, percent = 0.05):
     print "Total number of dead cells=",len(self.deadCells)
 
     numSegmentDeleted = 0
-    for cell in self.deadCells:
-      segmentsPerCell = list(self.connections.segmentsForCell(cell))
-      for segment in segmentsPerCell:
+
+    for cellIdx in self.deadCells:
+      # Destroy segments.  self.destroySegment() takes care of deleting synapses
+      for segment in self.connections.segmentsForCell(cellIdx):
         self.connections.destroySegment(segment)
         numSegmentDeleted += 1
 
     print "Total number of segments removed=", numSegmentDeleted
 
 
-  def activateCorrectlyPredictiveCells(self,
-                                       prevPredictiveCells,
-                                       prevMatchingCells,
-                                       activeColumns):
+  def burstColumn(self, connections, random, column, columnMatchingSegments,
+                  prevActiveCells, prevWinnerCells,
+                  numActivePotentialSynapsesForSegment, cellsPerColumn,
+                  maxNewSynapseCount, initialPermanence, permanenceIncrement,
+                  permanenceDecrement, learn):
```

Comment:
Wow! Overriding a class method with a traditional method... I've never thought of that. It only works because the base class calls `self.burstColumn`, not `cls.burstColumn`. This seems a little error-prone -- if the TemporalMemory is ever refactored to call `cls.burstColumn` it will break this (I think).

I think maybe we should change the base TemporalMemory to use regular methods. So the signature will be `burstColumn(self, column, columnMatchingSegments, prevActiveCells, prevWinnerCells)`, and it will immediately call a function `_burstColumn` with the really long argument list. Note this important point: we won't pass in `self.connections`, etc. to this method. The method will access the self and won't contain logic.

We'll will still have long argument lists, so it's not quite the same as https://github.com/numenta/nupic/issues/3309